File: practise/BinerySearchTree.go
package main

import "fmt"

type Data struct {
	val   int
	left  *Data
	right *Data
}

func main() {

	root := &Data{}

	addData(root, 100)
	addData(root, 95)
	addData(root, 80)
	addData(root, 101)

	// fmt.Println(root)
	// printData(root)

}

// func printData(da *Data) {
// 	if da.val != 0 {
// 		fmt.Println(da.val)
// 		if da.right != nil {
// 			printData(da.right)
// 		} else if da.right != nil {
// 			printData(da.left)
// 		}
// 	}
// }

func addData(da *Data, val int) {

	if da.val == 0 {

		da.val = val
		fmt.Println(da)
	} else {
		insertData(da, val)
	}
}

func insertData(da *Data, val int) {
	fmt.Println(val)
	if val > da.val {
		if da.right == nil {
			newData := Data{val: val}
			da.right = &newData
		} else {
			insertData(da.right, val)
		}
	} else {
		if da.left == nil {
			newData := Data{val: val}
			da.left = &newData
		} else {
			insertData(da.left, val)
		}
	}
}

File: practise/javascript/BinerySearchTree.js





let data={
    value:null,
    left:null,
    right:null
}




    function addData(val){
        if(data.value==null)
            {
                data.value=val
            }
            else{
                insertNode(data,val)
            }

    }

    function insertNode(root,val){
       
            if(root.value>val)
                {
                    if(root.left==null)
                        {
                            let data={
                                value:val,
                                left:null,
                                right:null
                            }
                            root.left=data
                        }
                        else{
                            insertNode(root.left,val)
                        }
                }
                else{
                    if(root.right==null)
                        {
                            let data={
                                value:val,
                                left:null,
                                right:null
                            }
                            root.right=data

                            // root.value=val
                            // root.left=null
                            // root.right=null
                        }else{
                            insertNode(root.right,val)
                        }

                }

    }


        addData(100)
        addData(90)
        addData(80)
        addData(85)
        addData(79)
        addData(110)
        addData(91)






        console.log(JSON.stringify(data));
